lib/galaxy/authnz/custos_authnz.py

```diff
@@ -436,6 +436,13 @@ def _get_well_known_uri_from_url(self, provider):
             # Remove potential trailing slash to avoid "//realms"
             base_url = base_url if base_url[-1] != "/" else base_url[:-1]
             return f"{base_url}/.well-known/openid-configuration"
+        if provider == "cilogon":
+            base_url = self.config["url"]
+            # backwards compatibility. CILogon URL is given with /authorize in the examples. not sure if
+            # this applies to the wild, but let's be safe here and remove the /authorize if it exists
+            # which will lead to the correct openid configuration
+            base_url = base_url if base_url.split("/")[-1] != "authorize" else "/".join(base_url.split("/")[:-1])
```

Comment:
Should we inform the admin when this happens via a log message?

Reply:
I believe that is a philosophical question on what the URL parameter in openid_backends_config should represent ;)

for Custos & Keycloak it is the base url, not the authorize endpoint (https://github.com/galaxyproject/galaxy/blob/77632581bc0a8a21b9279af4b86e793956edbbeb/lib/galaxy/config/sample/oidc_backends_config.xml.sample#L94 , https://github.com/galaxyproject/galaxy/blob/77632581bc0a8a21b9279af4b86e793956edbbeb/lib/galaxy/config/sample/oidc_backends_config.xml.sample#L133C1-L134C1) 

For CILogon it is the authorize endpoint:
https://github.com/galaxyproject/galaxy/blob/77632581bc0a8a21b9279af4b86e793956edbbeb/lib/galaxy/config/sample/oidc_backends_config.xml.sample#L114C19-L114C20

Ideally there shouldn't be a difference in what the URL parameter points to between providers. If you are planning to homogenise CILogon/Custos/Keycloak down the track we could throw a message and update the template.